controllers/factory/alertmanager/statefulset.go

```diff
@@ -452,6 +452,10 @@ func createDefaultAMConfig(ctx context.Context, cr *victoriametricsv1beta1.VMAle
 	switch {
 	// fetch content from user defined secret
 	case cr.Spec.ConfigSecret != "":
+		if cr.Spec.ConfigSecret != cr.ConfigSecretName() {
```

Comment:
I believe it should be `if cr.Spec.ConfigSecret == cr.ConfigSecretName() `?

Reply:
Good catch! I incorrectly copy-pasted line.